src/mindlessgen/prog/config.py

```diff
@@ -735,6 +737,66 @@ def __init__(self, config_file: str | Path | None = None):
         if config_file:
             self.load_from_toml(config_file)
 
+    def check_config(self, verbosity: int = 1) -> None:
+        """
+        Checks ConfigClass for any incompatibilities that are imaginable
+        """
+
+        # lower number of the available cores and the configured parallelism
+        num_cores = min(mp.cpu_count(), self.general.parallel)
+        if self.general.parallel > mp.cpu_count():
+            warnings.warn(
+                f"Number of cores requested ({self.general.parallel}) is greater "
+                + f"than the number of available cores ({mp.cpu_count()})."
+                + f"Using {num_cores} cores instead."
+            )
+
+        if num_cores > 1 and self.postprocess.debug:
+            # raise warning that debugging of postprocessing will disable parallelization
+            warnings.warn(
+                "Debug output might seem to be redundant due to the parallel processes "
+                + "with possibly similar errors in parallel mode. "
+                + "Don't be confused!"
+            )
+
+        if verbosity > 0:
```

Comment:
Switch the ordering between both if statements for this and all later checks: The ordering should be more like

- check 1
   - if verbosity > 1: print something
- check 2
   - if verbosity > X: print something or warning 

Reason is that we want to make the code consistent for cases in which you don't wanna print something but instead raise an error or similar.